### ISSUE TITLE
Crashes with DatePicker Year, Month, and Day parameters describe an un-representable DateTime

### ISSUE THREAD

**Issue by pracsec** [AvaloniaUI/Avalonia#10890]
**Describe the bug**
Certain input to the DatePicker results in "Year, Month, and Day parameters describe an un-representable DateTime" exception that crashes the program.

**To Reproduce**

1. Go to page with DatePicker.
2. Click on month to bring up menu.
3. Click on another month, but don't click on the check mark below.
4. Click off of the control.
5. Click on month again to trigger crash.

**Expected behavior**
Should not crash on interaction with DatePicker.

**Desktop (please complete the following information):**

- OS: Windows
- Version 0.10.19

**Additional context**

Here is the stack trace:

```
   at System.ThrowHelper.ThrowArgumentOutOfRange_BadYearMonthDay()
   at Avalonia.Controls.Primitives.DateTimePickerPanel.FormatContent(Int32 value, DateTimePickerPanelType panelType)
   at Avalonia.Controls.Primitives.DateTimePickerPanel.UpdateItems()
   at Avalonia.Controls.Primitives.DateTimePickerPanel.set_MaximumValue(Int32 value)
   at Avalonia.Controls.DatePickerPresenter.InitPicker()
   at Avalonia.Controls.DatePickerPresenter.set_Date(DateTimeOffset value)
   at Avalonia.Controls.DatePicker.OnFlyoutButtonClicked(Object sender, RoutedEventArgs e)
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)
   at Avalonia.Interactivity.EventRoute.RaiseEvent(IInteractive source, RoutedEventArgs e)
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)
   at Avalonia.Controls.Button.OnClick()
   at Avalonia.Controls.Button.OnPointerReleased(PointerReleasedEventArgs e)
   at System.Reactive.Subjects.Subject`1.OnNext(T value)
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)
   at Avalonia.Interactivity.EventRoute.RaiseEvent(IInteractive source, RoutedEventArgs e)
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)
   at Avalonia.Input.MouseDevice.MouseUp(IMouseDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties props, KeyModifiers inputModifiers)
   at Avalonia.Input.MouseDevice.ProcessRawEvent(RawPointerEventArgs e)
   at Avalonia.Input.InputManager.ProcessInput(RawInputEventArgs e)
   at Avalonia.Win32.WindowImpl.AppWndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)
   at Avalonia.Win32.WindowImpl.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)
   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)
   at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode)
   at SpecterInsight.UI.Program.Main(String[] args) in C:\\Users\\helpdesk\\Desktop\\Workspace\\repos\\SpecterInsight\\src\\Clients\\SpecterInsight.UI\\Program.cs:line 27
```

**Comment by SuperJMN** [comment]
Same issue here.

**Comment by mgnslndh** [comment]
I tried but could not reproduce this on Windows 11 from master 87ffbfc089bc68b491f8a33f621ab18d449136ab

Not sure what could have fixed it but perhaps this one 87ffbfc089bc68b491f8a33f621ab18d449136ab (#10370) which changes coercion of the Date property and also makes some changes to methods that are part of the exception stack trace.

**Comment by timunie** [comment]
closing as one user confirms it was solved. If someone can reproduce with latest nightly, ping me and I'll re-open.
